oximeter/timeseries-macro/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+extern crate proc_macro;
+
+use oximeter_impl::schema::SCHEMA_DIRECTORY;
+
+/// Generate code to use the timeseries from one target.
+///
+/// This macro accepts a single filename, which should be a file in the
+/// `oximeter/schema` subdirectory, containing a valid timeseries definition. It
+/// attempts parse the file and generate code used to produce samples from those
+/// timeseries. It will generate a submodule named by the target in the file,
+/// with a Rust struct for the target and each metric defined in the file.
+#[proc_macro]
+pub fn use_timeseries(
+    tokens: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    match syn::parse::<syn::LitStr>(tokens) {
+        Ok(token) => {
+            let filename = token.value();
+            let path = std::path::Path::new(SCHEMA_DIRECTORY).join(&filename);
```

Comment:
Not sure how much we care about this, but `.join()` is pretty flexible; e.g., if someone calls this with a filename of `/foo/bar/baz`, `SCHEMA_DIRECTORY` will be discarded entirely. If we want to _require_ that this be inside `SCHEMA_DIRECTORY`, we maybe want to call `filename.file_name()` to strip any parent directories or `../` escapes?

Reply:
Yeah, nice catch, thanks. I might choose to be even more conservative here at this point, just disallowing any slashes in the name at all, and / or checking that the `file_name()` output is the same as the input literal string itself. Thanks!

Reply:
Done in 6a7b8ae7f